document-store/src/main/java/org/hypertrace/core/documentstore/TypesafeDatastoreConfigAdapter.java

```diff
@@ -31,13 +34,32 @@ public DatastoreConfig convert(final Config config) {
 
       final MongoConnectionConfig overridingConnectionConfig =
           new MongoConnectionConfig(
-              connectionConfig.host(),
-              connectionConfig.port(),
+              connectionConfig.endpoints(),
               connectionConfig.database(),
               connectionConfig.credentials(),
-              connectionConfig.applicationName()) {
-            @Override
-            public ConnectionString toConnectionString() {
+              connectionConfig.applicationName(),
+              null) {
+            public MongoClientSettings toSettings() {
```

Comment:
What is change for `toSettings` ? Def looks same what is there in class `MongoConnectionConfig`.

Reply:
This is overriding the implementation in `MongoConnectionConfig`.
The original implementation is consuming the configs directly. And, the overridden implementation is generating a connection string and using it.